custom_components/deltasol/deltasolapi.py

```diff
@@ -43,40 +44,78 @@ def __parse_data(self, response):
         return data
 
 
+    def detect_mode(self):
+        if self.api_mode is not None:
+            return self.api_mode
+        
+        try:
+            dlxurl = f"http://{self.host}/dlx/"
+            _LOGGER.debug(f"Testing DLX {dlxurl}")
+            response = requests.request("GET", dlxurl)
+            _LOGGER.debug(f"DLX response {response}")
+            _LOGGER.debug(f"Detected DLX")
+            self.api_mode = 'dlx'
+            return self.api_mode
+
+        except ValueError:
+            _LOGGER.debug("Unable to detect DLX")
+
+        _LOGGER.debug("Testing KM2")
+        # find way to test for km2
```

Comment:
The old way was calling the km2 and if we can parse the result to json we assumed that is km2 and otherwise we went with dlx. Already made a comment about that before

Reply:
resolved by use of `GET /cgi-bin/get_resol_device_information`